src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1657,15 +1657,16 @@ bool QgsOgrProvider::addFeaturePrivate( QgsFeature &f, Flags flags )
     }
   }
 
-  QgsAttributes attrs = f.attributes();
+  QgsAttributes attributes = f.attributes();
+  const QgsFields &qgisFields { f.fields() };
```

Comment:
```suggestion
  const QgsFields qgisFields { f.fields() };
```

no need for a reference here -- QgsFields are implicitly shared, so copies are inexpensive.